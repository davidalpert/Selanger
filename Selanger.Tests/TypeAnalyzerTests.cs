using System;
using System.IO;
using System.Linq;
using ApprovalTests;
using NUnit.Framework;
using Selanger.Core;

namespace Selanger.Tests
{
    [TestFixture]
    public class TypeAnalyzerTests
    {
        [Test]
        public void CanGenerateTypeReport()
        {
            var asm = @".\nunit.framework.dll";
            var file = new FileInfo(asm);
            var reporter = new TypeAnalyzer();
            var result = reporter.Analyze(file);
            Approvals.VerifyAll(result, x => String.Format("{0},{1},{2},{3},{4}", 
                                                            x.Namespace,
                                                            x.Name, 
                                                            x.FileDirectory, 
                                                            x.FileName,
                                                            x.FileVersion
                                                           ));
        }

        [Test]
        public void CanGenerateTypeReportForListOfAssemblies()
        {
            var files = Directory.GetFiles(".", "*.dll",SearchOption.AllDirectories)
                                .Take(3)
                                .Select(f => new FileInfo(f))
                                .ToArray();

            var reporter = new TypeAnalyzer();

            var result = reporter.Analyze(files);

            Approvals.VerifyAll(result, x => String.Format("{0},{1},{2},{3},{4}", 
                                                            x.Namespace,
                                                            x.Name, 
                                                            x.FileDirectory, 
                                                            x.FileName,
                                                            x.FileVersion
                                                           ));
        }

        [Test]
        public void CanGenerateNamespaceReport()
        {
            var asm = @".\nunit.framework.dll";
            var file = new FileInfo(asm);
            var reporter = new TypeAnalyzer();
            var result = reporter.SummarizeNamespaces(file);
            Approvals.VerifyAll(result, x => String.Format("{0},{1},{2},{3}", 
                                                            x.AssemblyName, 
                                                            x.AssemblyVersion,
                                                            x.Namespace, 
                                                            x.Count
                                                           ));
        }

        [Test]
        public void CanGenerateNamespaceReportForListOfAssemblies()
        {
            var files = Directory.GetFiles(".", "*.dll",SearchOption.AllDirectories)
                                .Take(3)
                                .Select(f => new FileInfo(f))
                                .ToArray();

            var reporter = new TypeAnalyzer();

            var result = reporter.SummarizeNamespaces(files);

            Approvals.VerifyAll(result, x => String.Format("{0},{1},{2},{3}", 
                                                            x.Namespace, 
                                                            x.Count,
                                                            x.AssemblyName, 
                                                            x.AssemblyVersion
                                                           ));
        }
    }
}
