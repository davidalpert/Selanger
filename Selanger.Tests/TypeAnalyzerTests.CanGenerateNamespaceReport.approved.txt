nunit.framework.dll,2.6.3.13283,NUnit.Framework,86
nunit.framework.dll,2.6.3.13283,NUnit.Framework.Constraints,122
